Auth/Credentials/ICredential.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+namespace AeroGear.Mobile.Auth.Credentials
```

Comment:
I updated the structure of the module in PR #11. I think we should keep it consistent across all the modules. We either use the same nested folders in all the modules, or we don't use them at all.

@wojta look at this PR, it doesn't looks like we need to nested folder structure at all. So should we remove it from the core module as well (but we keep the same namespace)?